daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -951,12 +951,13 @@ private[lf] object SBuiltin {
         case SContractId(coid) => coid
         case v => crash(s"expected contract id, got: $v")
       }
-      val sigs = extractParties(args.get(2))
-      val templateObservers = extractParties(args.get(3))
-      val ctrls = extractParties(args.get(4))
-      val choiceObservers = extractParties(args.get(5))
+      val cached = onLedger.cachedContracts.get(coid).get
```

Comment:
```suggestion
      val cached = onLedger.cachedContracts(coid)
```
The `apply` method is defined on `Map`s to ultimately have the same effect as this, but the error message is more helpful:
```
scala> Map(1 -> "one")(2)
java.util.NoSuchElementException: key not found: 2
  at scala.collection.immutable.Map$Map1.apply(Map.scala:245)
  ... 32 elided

scala> Map(1 -> "one").get(2).get
java.util.NoSuchElementException: None.get
  at scala.None$.get(Option.scala:627)
  at scala.None$.get(Option.scala:626)
  ... 32 elided
```

Reply:
hah, I actually deliberately didn’t use `apply` to remind me that I need to emit a proper error message here since I find `.get` much easier to spot than an `apply` somewhere. Obviously I still forgot, will fix on Monday :slightly_smiling_face:

Reply:
2fbd75434cb8198a46429a65d02a7aa8639e6281